### PR TITLE
#20: Add Firebase Authentication

### PR DESCRIPTION
**Pull request by ShashankKasera**

### CODE REVIEW

**Comment by ShashankKasera** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Having a proper description on PR gives more clarity to the reviewer. Please ensure that the PR has a neatly written description explaining the purpose of this change request.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Let's also assign this pull request to someone for review.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems">Danger Kotlin</a> against 1d37a54c82daa60bd8693b5c6c0463fe964e6f1c
</p>

File: authentication/src/main/java/com/example/authentication/LoginActivity.kt [hunk 1]
```diff
@@ -1,13 +1,76 @@
...
+
+        emailAddress = findViewById(R.id.et_Email)
+        password = findViewById(R.id.et_Password)
+        loginBtn = findViewById(R.id.tv_login)
+
+        loginBtn.setOnClickListener {
+            sEmailAddress = emailAddress.text.toString().trim()
+            sPassword = password.text.toString().trim()
+
+            Log.i("wjlogijl", "onCreate: $sEmailAddress $sPassword")
```
Comment (mayankkasera) [inline authentication/src/main/java/com/example/authentication/LoginActivity.kt:39]: Remove this log.

File: authentication/src/main/java/com/example/authentication/LoginActivity.kt [hunk 2]
```diff
@@ -1,13 +1,76 @@
...
+            auth.signInWithEmailAndPassword(sEmailAddress, sPassword)
+                .addOnCompleteListener(this) { task ->
+                    if (task.isSuccessful) {
+                        // Sign in success, update UI with the signed-in user's information
+                        Log.d("wjlogijl", "signInWithEmail:success")
```
Comment (mayankkasera) [inline authentication/src/main/java/com/example/authentication/LoginActivity.kt:44]: Remove all unused comments.

File: authentication/src/main/java/com/example/authentication/LoginActivity.kt (hunk 1 above)
Comment (ShashankKasera) [reply authentication/src/main/java/com/example/authentication/LoginActivity.kt]: done

File: authentication/src/main/java/com/example/authentication/LoginActivity.kt (hunk 2 above)
Comment (ShashankKasera) [reply authentication/src/main/java/com/example/authentication/LoginActivity.kt]: done

**Pull request merged**